android/engine/src/main/java/org/smartregister/fhircore/engine/ui/appsetting/AppSettingViewModel.kt

```diff
@@ -98,11 +101,29 @@ constructor(
           val resourceUrlPath = entry.key + "?${Composition.SP_RES_ID}=$ids"
           fhirResourceDataSource.loadData(resourceUrlPath).entry.forEach {
             defaultRepository.create(it.resource)
+
+            if (it.resource is Binary) {
+              val binary = it.resource as Binary
+              binary.data.decodeToString().decodeBase64()!!.string(Charset.defaultCharset()).let {
+                val config =
+                  it.tryDecodeJson<RegisterConfiguration>()
+                    ?: it.tryDecodeJson<ProfileConfiguration>()
+
+                when (config) {
+                  is RegisterConfiguration ->
+                    config.fhirResource.dependentResourceTypes(patientRelatedResourceTypes)
+                  is ProfileConfiguration ->
+                    config.fhirResource.dependentResourceTypes(patientRelatedResourceTypes)
```

Comment:
because you only need a single branch, this is better suited for an if, or at least combine the conditions

```suggestion
                when (config) {
                  is ProfileConfiguration, RegisterConfiguration ->
                    config.fhirResource.dependentResourceTypes(patientRelatedResourceTypes)
```

Reply:
I would change it to if else. The reason is it does smart cast when `is` is used i.e. smart cast to RegisterConfiguration. Unfortunately the properties for both register/profile config are similar that why the code looks repeating. Ideally we should have used a parent class for both classes for same properties. 
Using comma separated values it gives compile error 'Incompatible types: ProfileConfiguration.Companion and Configuration?'